diff --git a/myelements/elements.py b/myelements/elements.py
index a95eac1..e7ab24d 100644
--- a/myelements/elements.py
+++ b/myelements/elements.py
@@ -495,6 +495,7 @@ class Elements:
         bodylist = []
         for body in self.world.bodies:
             if body == self.world.groundBody:
+                body.userData = {"saveid": 0}
                 continue
             body.userData["saveid"] = save_id_index  # set temporary data
             save_id_index += 1
