diff --git a/tools.py b/tools.py
index 78108fe..3f3c4b3 100644
--- a/tools.py
+++ b/tools.py
@@ -98,7 +98,8 @@ class Tool(object):
                 elif event.action == 'clear_all':
                     # Get bodies and destroy them too
                     for body in self.game.world.world.bodies:
-                        self.game.world.world.DestroyBody(body)
+                        if body != self.game.world.world.groundBody:
+                            self.game.world.world.DestroyBody(body)
 
                     # Add ground, because we destroyed it before
                     self.game.world.add.ground()
