diff --git a/tools.py b/tools.py
index 31af72e..728c5e7 100644
--- a/tools.py
+++ b/tools.py
@@ -615,7 +615,7 @@ class GrabTool(Tool):
                     if self.pm_mode_active:
                         pos = event.pos
                         body = find_body(self.game.world, pos)
-                        if body is not None:
+                        if hasattr(self._moving_pm, 'enableMotor') and body is not None:
                             if self._moving_pm.enableMotor:
                                 self.game.world.add.motor(
                                     body, pos,

