diff --git a/activity.py b/activity.py
index 18c3593..d3d0e72 100644
--- a/activity.py
+++ b/activity.py
@@ -60,19 +60,21 @@ from collabwrapper import CollabWrapper
 import tools
 from physics import PhysicsGame
 
+from sugarapp.widgets import ExtendedActivityToolbarButton
+from sugarapp.widgets import SugarCompatibleActivity
+
 # For some reason increasing FPS decreases execution speed :/
 SLOWEST_FPS = 90
 SLOW_FPS = 70
 FAST_FPS = 50
 
 
-class PhysicsActivity(Activity):
+class PhysicsActivity(SugarCompatibleActivity):
 
     def __init__(self, handle):
-        Activity.__init__(self, handle)
+        SugarCompatibleActivity.__init__(self, handle)
         self._collab = CollabWrapper(self)
         self._collab.message.connect(self.__message_cb)
-        self.metadata['mime_type'] = 'application/x-physics-activity'
         self.add_events(Gdk.EventMask.ALL_EVENTS_MASK |
                         Gdk.EventMask.VISIBILITY_NOTIFY_MASK)
 
@@ -141,7 +143,7 @@ class PhysicsActivity(Activity):
         self.set_toolbar_box(toolbar_box)
         toolbar_box.show()
 
-        activity_button = ActivityToolbarButton(self)
+        activity_button = ExtendedActivityToolbarButton(self)
         toolbar_box.toolbar.insert(activity_button, -1)
         activity_button.show()
 
@@ -202,22 +204,9 @@ class PhysicsActivity(Activity):
         toolbar_box.toolbar.insert(stop, -1)
         stop.show()
 
-        separator = Gtk.SeparatorToolItem()
-        activity_button.props.page.insert(separator, -1)
-        separator.show()
-
-        export_json = ToolButton('save-as-json')
-        export_json.set_tooltip(_('Export tracked objects to journal'))
-        export_json.connect('clicked', self._export_json_cb)
-        activity_button.props.page.insert(export_json, -1)
-        export_json.show()
-
-        load_project = ToolButton('load-project')
-        load_project.set_tooltip(_('Load project from journal'))
-        load_project.connect('clicked', self._load_project)
-        activity_button.props.page.insert(load_project, -1)
-        load_project.show()
-
+        self.set_toolbar_box(toolbar_box)
+        toolbar_box.show_all()
+        create_toolbar.set_expanded(True)
         return toolbar_box
 
     def __color_notify_cb(self, button, pdesc):
@@ -282,7 +271,6 @@ class PhysicsActivity(Activity):
         return (color.red // 256, color.green // 256, color.blue // 256)
 
     def can_close(self):
-        self.preview = self.get_preview()
         self.game.running = False
         return True
 
@@ -484,8 +472,6 @@ class PhysicsActivity(Activity):
 
     def _export_json_cb(self, button):
         jobject = datastore.create()
-        jobject.metadata['title'] = _('Physics export')
-        jobject.metadata['mime_type'] = 'text/plain'
 
         tmp_dir = os.path.join(self.get_activity_root(), 'instance')
         fd, file_path = tempfile.mkstemp(dir=tmp_dir)
@@ -619,15 +605,6 @@ class PhysicsActivity(Activity):
     def send_event(self, text):
         self._collab.post(dict(action='text', text=text))
 
-    def _load_project(self, button):
-        chooser = ObjectChooser(parent=self)
-        result = chooser.run()
-        if result == Gtk.ResponseType.ACCEPT:
-            dsobject = chooser.get_selected_object()
-            file_path = dsobject.get_file_path()
-            self.__load_game(file_path, True)
-            chooser.destroy()
-
     def __load_game(self, file_path, journal=False):
         confirmation_alert = ConfirmationAlert()
         confirmation_alert.props.title = _('Are You Sure?')
