diff --git a/activity.py b/activity.py
index ca761d8..f5c074c 100644
--- a/activity.py
+++ b/activity.py
@@ -58,6 +58,7 @@ from collabwrapper import CollabWrapper
 import tools
 from physics import PhysicsGame
 
+from sugarapp.helpers import PrimaryMonitor
 from sugarapp.widgets import ExtendedActivityToolbarButton
 from sugarapp.widgets import SugarCompatibleActivity
 
@@ -89,8 +90,8 @@ class PhysicsActivity(SugarCompatibleActivity):
         self._notebook = Gtk.Notebook(show_tabs=False)
         self._notebook.add(self.game.canvas)
 
-        w = Gdk.Screen.width()
-        h = Gdk.Screen.height() - 2 * GRID_CELL_SIZE
+        w = PrimaryMonitor.width()
+        h = PrimaryMonitor.height() - 2 * GRID_CELL_SIZE
 
         self.game.canvas.set_size_request(w, h)
 
@@ -116,8 +117,8 @@ class PhysicsActivity(SugarCompatibleActivity):
         ''' Screen size has changed '''
         self.write_file(os.path.join(
                         activity.get_activity_root(), 'data', 'data'))
-        w = Gdk.Screen.width()
-        h = Gdk.Screen.height() - 2 * GRID_CELL_SIZE
+        w = PrimaryMonitor.width()
+        h = PrimaryMonitor.height() - 2 * GRID_CELL_SIZE
         pygame.display.set_mode((w, h),
                                 pygame.RESIZABLE)
         self.read_file(os.path.join(
@@ -645,7 +646,7 @@ class PhysicsActivity(SugarCompatibleActivity):
             self._sample_window.set_border_width(4)
 
             self._sample_window.set_size_request(
-                Gdk.Screen.width() / 2, Gdk.Screen.height() / 2)
+                PrimaryMonitor.width() / 2, PrimaryMonitor.height() / 2)
             self._sample_window.show()
 
             store = Gtk.ListStore(GdkPixbuf.Pixbuf, str)

